components/rangeslider/script.js

```diff
@@ -0,0 +1,460 @@
+import components from 'coherent-gameface-components';
+import verticalTemplate from './templates/vertical.html';
+import verticalTemplateTwoHandles from './templates/verticalTwoHandles.html';
+import horizontalTemplate from './templates/horizontal.html';
+import horizontalTemplateTwoHandles from './templates/horizontalTwoHandles.html';
+import { orientationUnitsNames } from './orientationUnitsNames';
+
+const ORIENTATIONS = ['vertical', 'horizontal'];
+const SPACE_BETWEEN_GRID_POLS = 10;
+
+/**
+ * @class Rangeslider.
+ * Rangeslider component, allows you to specify a numeric value by using a slider.
+ * It must be no less than a given value, and no more than another given value.
+ */
+class Rangeslider extends HTMLElement {
+    /**
+     * Sets the minimum value of the slider
+     * @param {number} value
+     */
+    set min(value) {
+        this._min = value;
+    }
+
+    /**
+     * Gets the minimum value of the slider
+     * @returns {number}
+     */
+    get min() {
+        return this._min;
+    }
+
+    /**
+     * Sets the maximum value of the slider
+     * @param {number} value
+     */
+    set max(value) {
+        this._max = value;
+    }
+
+    /**
+     * Gets the maximum value of the slider
+     * @returns {number}
+     */
+    get max() {
+        return this._max;
+    }
+
+    /**
+     * Converts a value to percent in a range
+     * @param {number} value - the value to be converted
+     * @param {number} min - the minimum value of the range
+     * @param {number} max - the maximum number of the range
+     * @returns {number} - returns the value in percent
+     */
+    static valueToPercent(value, min, max) {
+        return (value * 100) / (max - min);
+    }
+
+    /**
+     * Restricts a given value in a range
+     * @param {number} val - the value to be restricted
+     * @param {number} min - the minimum value of the range
+     * @param {number} max - the maximum number of the range
+     * @returns {number}
+     */
+    static clamp(val, min, max) {
+        return Math.min(Math.max(val, min), max);
+    }
+
+    constructor() {
+        super();
+
+        this.onMouseDown = this.onMouseDown.bind(this);
+        this.onMouseMove = this.onMouseMove.bind(this);
+        this.onMouseUp = this.onMouseUp.bind(this);
+    }
+
+    /**
+     * Called when the element was attached to the DOM.
+     */
+    connectedCallback() {
+        //if an array is passed these are the values of the array
+        this._values = this.hasAttribute('values') ? JSON.parse(this.getAttribute('values')) : null;
+
+        //if an array is passed
+        this.hasValuesArray = Array.isArray(this._values) && this._values.length > 0;
```

Comment:
Since the other instances are `isValuesArray`, rename this to it because it currently isn't used anywhere and probably the `isValuesArray` uses below cause an unexpected behavior.